src/components/ListItem.jsx

```diff
@@ -1,8 +1,29 @@
 import { updateItem } from '../api';
-import { withinTwentyFourHours } from '../utils';
+import { withinTwentyFourHours, getDaysBetweenDates } from '../utils';
 import './ListItem.css';
 
 export function ListItem({ itemData, listToken }) {
+	// determine item's status based on dateNextPuchased and dateLastPurchased
+	const stringSetter = () => {
+		const today = new Date();
+		const daysUntilNextPurchase = getDaysBetweenDates(
+			itemData.dateNextPurchased.toDate(),
+			today,
+		);
+
+		if (
+			itemData.dateLastPurchased !== null &&
+			getDaysBetweenDates(today, itemData.dateLastPurchased.toDate()) >= 60
+		)
+			return ['(inactive)', 'item-inactive'];
+		if (daysUntilNextPurchase < 0) return ['(overdue)', 'item-overdue'];
+		if (daysUntilNextPurchase <= 7) return ['(soon)', 'item-short'];
+		if (daysUntilNextPurchase < 29) return ['(kind of soon)', 'item-mid'];
+		return ['(not soon)', 'item-long'];
+	};
+
+	const [statusString, statusClass] = stringSetter();
```

Comment:
Nice work setting this up! It's a neat way of organizing the class names and text values.